### PR TITLE
Phong color mapping for scalar field data, refactor tutorials 

### PR DESCRIPTION
**Pull request by alecjacobson**
We'll need to update the tutorial screenshots to match these better visualizations.

# Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [x] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [ ] This is a minor change.

### CODE REVIEW

File: include/igl/opengl/ViewerData.h [hunk 1]
```diff
@@ -96,6 +95,26 @@ class ViewerData
...
     const Eigen::Matrix<unsigned char,Eigen::Dynamic,Eigen::Dynamic>& A);
+  // Set pseudo-colorable scalar data associated with the mesh.
+  //
+  // Inputs:
+  //   caxis_min  caxis minimum bound
+  //   caxis_max  caxis maximum bound
+  //   D  #V by 3 list of colors
+  //
+  // To-do: support #F by 1 per-face data
+  IGL_INLINE void set_data(
```
Comment (jdumas) [inline include/igl/opengl/ViewerData.h]: In `igl::colormap` we pass `minZ` and `maxZ` after the `Z` data. Should we do the same here for consistency?

File: include/igl/opengl/ViewerData.cpp [hunk 2]
```diff
@@ -260,6 +264,44 @@ IGL_INLINE void igl::opengl::ViewerData::set_texture(
...
 
+IGL_INLINE void igl::opengl::ViewerData::set_data(
+  const double caxis_min,
+  const double caxis_max,
+  const Eigen::VectorXd & D)
+{
+  if(!show_texture)
+  {
+    Eigen::MatrixXd CM;
+    igl::parula(Eigen::VectorXd::LinSpaced(21,0,1).eval(),false,CM);
```
Comment (jdumas) [inline include/igl/opengl/ViewerData.cpp]: We should add an option `igl::ColorMapType` to specify the colormap. Also, our colormaps are defined using 256 samples in `igl/colormap.cpp`, so we should use the same number of samples here.

Comment (alecjacobson) [reply include/igl/opengl/ViewerData.cpp]: see other comment. 21 is to create identifiable intervals which helps with scalar field visualization.

File: include/igl/opengl/ViewerData.h (hunk 1 above)
Comment (alecjacobson) [reply include/igl/opengl/ViewerData.h]: agreed.

File: include/igl/opengl/ViewerData.h [hunk 3]
```diff
@@ -96,6 +95,26 @@ class ViewerData
     const Eigen::Matrix<unsigned char,Eigen::Dynamic,Eigen::Dynamic>& G,
     const Eigen::Matrix<unsigned char,Eigen::Dynamic,Eigen::Dynamic>& B,
     const Eigen::Matrix<unsigned char,Eigen::Dynamic,Eigen::Dynamic>& A);
+  // Set pseudo-colorable scalar data associated with the mesh.
+  //
+  // Inputs:
+  //   caxis_min  caxis minimum bound
+  //   caxis_max  caxis maximum bound
+  //   D  #V by 3 list of colors
```
[the remark this answers is not in the document]
Comment (alecjacobson) [reply include/igl/opengl/ViewerData.h]: agreed

**Pull request merged**